scripts/make_video.py

```diff
@@ -0,0 +1,154 @@
+    args = parser.parse_args()
+
+    # read config file
+    if args.cfg is not None:
+        with open(args.cfg) as f:
+            cfg = yaml.safe_load(f)
+        for key, val in cfg.items():
+            if hasattr(args, key):
+                setattr(args, key, val)
+            else:
+                raise ValueError(f'bad field in config file: {key}')
+
+    # check for prompts
+    if args.prompts is None:
+        raise ValueError('no prompt provided')
+    if args.seeds is None:
+        args.seeds = [random.getrandbits(16) for _ in args.prompts]
+
+    # check audio arguments
+    if args.audio_filepath is not None and args.audio_offsets is None:
+        raise ValueError('must provide audio_offsets when providing '
+                         'audio_filepath')
+    if args.audio_offsets is not None and args.audio_filepath is None:
+        raise ValueError('must provide audio_filepath when providing '
+                         'audio_offsets')
+
+    # check lengths
+    if args.audio_offsets is not None:
+        if not len(args.prompts) == len(args.seeds) == len(args.audio_offsets):
+            raise ValueError('prompts, seeds and audio_offsets must have same '
+                             f'length, got lengths {len(args.prompts)}, '
+                             f'{len(args.seeds)} and '
+                             f'{len(args.audio_offsets)} respectively')
+    else:
+        if not len(args.prompts) == len(args.seeds):
+            raise ValueError('prompts and seeds must have same length, got '
+                             f'lengths {len(args.prompts)} and '
+                             f'{len(args.seeds)} respectively')
+
+    # set num_interpolation_steps
+    if args.audio_offsets is not None \
+            and args.num_interpolation_steps is not None:
+        raise ValueError('cannot provide both audio_offsets and '
+                         'num_interpolation_steps')
+    elif args.audio_offsets is not None:
+        args.num_interpolation_steps = [
+            (b-a)*args.fps for a, b in zip(
+                args.audio_offsets, args.audio_offsets[1:]
+            )
+        ]
+    elif args.num_interpolation_steps is not None \
+            and not len(args.num_interpolation_steps) == len(args.prompts)-1:
+        raise ValueError('num_interpolation_steps must have length '
+                         f'len(prompts)-1, got '
+                         f'{len(args.num_interpolation_steps)} != '
+                         f'{len(args.prompts)-1}')
+    else:
+        args.num_interpolation_steps = args.fps*10  # 10 second video
+
+    return args
+
+
+def main():
+    parser = init_arg_parser()
+    args = parse_args(parser)
+
+    pipe = StableDiffusionWalkPipeline.from_pretrained(
+        args.checkpoint_id,
+        torch_dtype=torch.float16,
+        revision="fp16",
+        feature_extractor=None,
+        safety_checker=None,
+    ).to("cuda")
+    pipe.scheduler = DPMSolverMultistepScheduler.from_config(
```

Comment:
Hardcoding this likely bad idea too

Reply:
Oops that slipped my mind, these should be options too